src/util/typing.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Helper function for cases in which we know it's safe to apply a Ref
+ * that is a different Element type than the particular element expects.
+ *
+ * For example, the following will cause a type-checking error:
+ *
+ * param {Object} props
+ *   prop {import('preact').Ref<HTMLElement>} props.myRef
+ *
+ * function MyThing({ myRef }) {
+ *   return <div ref={myRef}>Hi there</div>;
+ * }
+ *
+ * The <div> wants {Ref<HTMLDivElement>}, while the component accepts the more
+ * generic {Ref<HTMLElement>}. TS' invariance in this case gets in our way.
```

Comment:
This commit has been removed from this branch because the need was satisfied by another bit of work—you can see more specific conversation about that here: https://github.com/hypothesis/frontend-shared/pull/172